src/main/webapp/app/lecture/pdf-preview/pdf-preview.component.ts

```diff
@@ -0,0 +1,203 @@
+        return overlay;
+    }
+
+    private resizeCanvasBasedOnContainer = () => {
+        if (this.isEnlargedView && this.enlargedCanvas) {
+            this.updateEnlargedCanvas(this.pdfContainer.nativeElement.querySelectorAll('.pdf-page-container canvas')[this.currentPage - 1]);
+        }
+    };
+
+    displayEnlargedCanvas(originalCanvas: HTMLCanvasElement, pageIndex: number) {
+        this.isEnlargedView = true;
+        this.currentPage = pageIndex;
+        this.updateEnlargedCanvas(originalCanvas);
+        this.toggleBodyScroll(true);
+    }
+
+    private updateEnlargedCanvas(originalCanvas: HTMLCanvasElement) {
+        requestAnimationFrame(() => {
+            if (this.isEnlargedView) {
+                const enlargedCanvas = this.enlargedCanvas.nativeElement;
+                const context = enlargedCanvas.getContext('2d');
+
+                const containerWidth = this.pdfContainer.nativeElement.clientWidth;
+                const containerHeight = this.pdfContainer.nativeElement.clientHeight;
+
+                const scaleX = containerWidth / originalCanvas.width;
+                const scaleY = containerHeight / originalCanvas.height;
+                const scaleFactor = Math.min(scaleX, scaleY);
+
+                enlargedCanvas.width = originalCanvas.width * scaleFactor;
+                enlargedCanvas.height = originalCanvas.height * scaleFactor;
+
+                context.clearRect(0, 0, enlargedCanvas.width, enlargedCanvas.height);
+                context.drawImage(originalCanvas, 0, 0, enlargedCanvas.width, enlargedCanvas.height);
+
+                enlargedCanvas.parentElement.style.top = `${this.pdfContainer.nativeElement.scrollTop}px`;
+                enlargedCanvas.style.position = 'absolute';
+                enlargedCanvas.style.left = `${(containerWidth - enlargedCanvas.width) / 2}px`;
+                enlargedCanvas.style.top = `${(containerHeight - enlargedCanvas.height) / 2}px`;
+            }
+        });
+    }
+
+    handleKeyboardEvents = (event: KeyboardEvent) => {
+        if (this.isEnlargedView) {
+            if (event.key === 'ArrowRight' && this.currentPage < this.totalPages) {
+                this.navigatePages('next');
+            } else if (event.key === 'ArrowLeft' && this.currentPage > 1) {
+                this.navigatePages('prev');
+            }
+        }
+    };
+
+    closeEnlargedView() {
+        this.isEnlargedView = false;
+        this.toggleBodyScroll(false);
+    }
+
+    toggleBodyScroll(disable: boolean): void {
+        this.pdfContainer.nativeElement.style.overflow = disable ? 'hidden' : 'auto';
+    }
+
+    closeIfOutside(event: MouseEvent): void {
+        const target = event.target as HTMLElement;
+        const enlargedCanvas = this.enlargedCanvas.nativeElement;
+
+        if (target.classList.contains('enlarged-container') && target !== enlargedCanvas) {
+            this.closeEnlargedView();
+        }
+    }
+
+    navigatePages(direction: string) {
```

Comment:
I think it would be good to add a custom type for the possible directions

Reply:
Good idea, added